scripts/install-macos.sh

```diff
@@ -0,0 +1,268 @@
+    mkdir -p "$GOROOT"
+
+    tar -C "$GOROOT" --strip-components=1 -xzf "$TEMP_DIRECTORY/go.tar.gz"
+
+    # GOROOT=${GOROOT}
+    # PATH=$GOROOT/bin:$PATH
+    # GOPATH=$GOPATH
+    # PATH=$GOPATH:$PATH
+
+    mkdir -p "${GOPATH}/"{src,pkg,bin}
+    echo -e "\nGo $VERSION was installed into $GOROOT.\nMake sure to relogin into your shell or run:"
+    echo -e "\n\ restart script to update your environment variables."
+    echo "Tip: Opening a new terminal window usually just works. :)"
+    rm -f "$TEMP_DIRECTORY/go.tar.gz"
+}
+
+# Once we have Golang; finish the install processes inside the development directory to avoid errors (Glide)
+function cloneIntoDir() {
+    echo "Setting up Kelp folders in the Golang working directory"
+    
+    # check github.com/stellar/kelp
+    if [ -d "$GOPATH/github.com/stellar/kelp" ]; then
+        echo "Kelp dir exists, no need to clone."
+    else
+        # pwd
+        mkdir -p $GOPATH/github.com/stellar/kelp
+        echo "Cloning Kelp into $GOPATH/src/github.com/stellar/kelp"
+        git clone https://github.com/stellar/kelp.git $GOPATH/src/github.com/stellar/kelp
+    fi
+
+    cd $GOPATH/src/github.com/stellar/kelp
+}
+
+# After Golang install Glide
+function installGlide() {
+    echo "Installing Glide."
+    if curl --version; then
+        curl https://glide.sh/get | sh
+    elif wget --version; then
+        wget https://glide.sh/get | sh
+    else
+        echo "curl and wget are not available, install glide manually https://github.com/Masterminds/glide"
+        exit
+    fi
+
+    glide install
+}
+
+# After Glide install Astilectron
+function installAstilectron() {
+    go get -u github.com/asticode/go-astilectron-bundler/... 
+    go install github.com/asticode/go-astilectron-bundler/astilectron-bundler
+}
+
+function postGoInstall() {
+    cloneIntoDir
+    isGlide
+    installAstilectron
+
+    checkDeps
+    echo "Remember, PostgreSQL must be running to store data."
+    echo "Remember, Docker and CCXT must be configured for the expanded set of priceFeeds and orderbooks."
+    echo "Run everything related to Kelp in BASH shell only."
+    echo "All dependencies successfully installed. Run build script from $GOPATH/src/github.com/stellar/kelp/scripts/build.sh"
+}
+
+# Utility function for checking dependency statuses
+function checkDeps(){
+    isNode
+    if [ $OK ]; then
+        echo "Node `node -v` is installed"
+    else
+        echo "Node is not installed"
+    fi
+
+    isYarn
+    if [ $OK ]; then
+        echo "Yarn `yarn -v` is installed"
+    else
+        echo "Yarn is not installed"
+    fi
+
+    isGit
+    if [ $OK ]; then
+        echo "`git version` is installed"
+    else
+        echo "Git is not installed"
+    fi
+
+    isGo
+    if $OK; then
+        echo "`go version` is installed"
+    else
+        echo "Goland is not installed"
+    fi
+}
+
+
+#**********************
+# Execute the functions
+#**********************
+cd
+
+isNode
+if [ $OK ]; then
```

Comment:
maybe we can call `checkDeps` here. so it executes all of these checks and we don't need this duplicate code.
then we can reuse the $OK variable and if it is false we can simply to an exit 1.

Example:
```
#checkDeps will set the OK variable to false if at least 1 dependency is missing
checkDeps
if [ $OK ]; then
    # do nothing
else
    exit 1
fi
``` 

Since we don't want to exit in the case where Golang does not exist, we can inline the go check logic from the `checkDeps` function into `postGoInstall()` right after we call `checkDeps` (line 190)